CMakeLists.txt

```diff
@@ -19,44 +19,60 @@ endif()
 # SDL2 Dependency
 find_package(SDL2)
 
-# Base FAudio Sources
-set(FAUDIO_SRC
-	src/F3DAudio.c
-	src/FAudio.c
-	src/FAudio_internal.c
-	src/FAudio_internal_simd.c
-	src/FAudioFX_reverb.c
-	src/FAudioFX_volumemeter.c
-	src/FACT.c
-	src/FACT3D.c
-	src/FACT_internal.c
-	src/FAPOBase.c
-	src/FAPOFX.c
-	src/FAPOFX_eq.c
-	src/FAPOFX_masteringlimiter.c
-	src/FAPOFX_reverb.c
-	src/FAPOFX_echo.c
-	src/FAudio_platform_sdl2.c
+# Source lists
+add_library(FAudio SHARED
+    src/F3DAudio.c
+    src/FAudio.c
+    src/FAudio_internal.c
+    src/FAudio_internal_simd.c
+    src/FAudioFX_reverb.c
+    src/FAudioFX_volumemeter.c
+    src/FACT.c
+    src/FACT3D.c
+    src/FACT_internal.c
+    src/FAPOBase.c
+    src/FAPOFX.c
+    src/FAPOFX_eq.c
+    src/FAPOFX_masteringlimiter.c
+    src/FAPOFX_reverb.c
+    src/FAPOFX_echo.c
+    src/FAudio_platform_sdl2.c
 )
+# include directory for other targets to consume
+target_include_directories(FAudio INTERFACE src)
 
 # FFmpeg Support
 if(FFMPEG)
 	# TODO: Add libavcodec/libavutil as dependencies
-	add_definitions(-DHAVE_FFMPEG=1)
-	list(APPEND FAUDIO_SRC src/FAudio_ffmpeg.c)
+    # add source file to FAudio target
+    target_sources(FAudio PRIVATE src/FAudio_ffmpeg.c)
+    target_compile_definitions(FAudio PRIVATE HAVE_FFMPEG=1)
 endif(FFMPEG)
 
 # XNA_Song Support
 if(XNASONG)
-	list(APPEND FAUDIO_SRC src/XNA_Song.c)
+    # add source file to FAudio target
+    target_sources(FAudio PRIVATE src/XNA_Song.c)
 endif(XNASONG)
 
-# FAudio Shared Library
-add_library(FAudio SHARED ${FAUDIO_SRC})
-include_directories(FAudio ${SDL2_INCLUDE_DIRS})
-target_link_libraries(FAudio ${SDL2_LIBRARY})
+# link SDL2 library
+find_package(SDL2 CONFIG REQUIRED)
+if (TARGET SDL2::SDL2)
+    message(STATUS "using TARGET SDL2::SDL2")
+    target_link_libraries(FAudio PUBLIC SDL2::SDL2)
+elseif (TARGET SDL2)
+    message(STATUS "using TARGET SDL2")
+    target_link_libraries(FAudio PUBLIC SDL2)
+else()
+    message(STATUS "no TARGET SDL2::SDL2, or SDL2, using variables")
+    target_include_directories(FAudio PUBLIC ${SDL2_INCLUDE_DIRS})
+    target_link_libraries(FAudio PUBLIC ${SDL2_LIBRARIES})
+endif()
 
 # TODO: utils/
```

Comment:
This should take care of the utils TODO. Are we able to folder the other CMakeLists.txt into this one? I try not to have nested CMake files unless the projects are actually separate from one another in a significant way.

Other than that, spaces->tabs should wrap this up!

Reply:
sure, I don't like to have huge CMakeLists files, but it is your repo, so your wish is my command :)

Reply:
I remember why I like subfolder cmake lists files. I now have to add `utils/` before every source file